src/ngraph/pass/constant_folding.cpp

```diff
@@ -234,6 +234,12 @@ void pass::ConstantFolding::construct_constant_reshape()
                          fold_constant_reshape<double>(constant_match, reshape_match, func));
             return true;
         }
+        else if (type == element::bf16)
+        {
+            replace_node(m.get_match_root(),
+                         fold_constant_reshape<int16_t>(constant_match, reshape_match, func));
```

Comment:
Should be okay since internally this is just moving 16-bit-whatevers around, but does it work to supply `<ngraph::bfloat16>` instead?

Reply:
wasn't aware of `<ngraph::bfloat16>`. I can take a closer look.